kolibri/core/assets/src/heartbeat.js

```diff
@@ -0,0 +1,58 @@
+const logging = require('kolibri.lib.logging').getLogger(__filename);
+
+class HeartBeat {
+  constructor(kolibri) {
+    this.kolibri = kolibri;
+    // Do this to have a consistent callback that has 'this' properly bound
+    // but can be repeatedly referenced to add and remove listeners.
+    this.setActiveCallback = this.setActive.bind(this);
+    this.setInactive();
+    this.start();
+  }
+  start() {
+    logging.debug('Starting heartbeat');
+    this.setActivityListeners();
+    this.beat();
+  }
+  setActivityListeners() {
+    this.events.forEach((event) => {
+      document.addEventListener(event, this.setActiveCallback, true);
+    });
+  }
+  clearActivityListeners() {
+    this.events.forEach((event) => {
+      document.removeEventListener(event, this.setActiveCallback, true);
+    });
+  }
+  setActive() {
+    this.active = true;
```

Comment:
This might be a pre-optimization? I'm curious if there would be noticeable performance gains by removing and re-adding the listeners, since all it does is set a boolean.

Also, perhaps unnecessary to make `HeartBeat` a class, since there most likely would never be more than once instance? Could potentially do just a module with internal state and a setup method. E.g. [something like this](https://gist.github.com/indirectlylit/affec61639f616dce4c1b32d2e7779ee) also does away with all the `bind` logic.